### DISCUSSION TITLE
Add GraphQL directive for strip_tags

### DISCUSSION THREAD

**Discussion by arthurdewolf** [craftcms/cms#9971] [Ideas] [answered]
Add a directive in GraphQL for stripping HTML tags, similar to PHP's `strip_tags` function, where you can pass it allowed tags.

Example:

```
{
entries (section: "news") {
  title
  ... on news_article_Entry {
    headline @strip_tags(allowed: "<a><strong>")
  }
}
```

We're using headless Craft with Nuxt.js and running into an issue where content editors are putting heading tags into a Redactor field that is outputted into another heading tag in the frontend, causing the JS to break. It would be nice if we could strip these tags out right in the GQL query.

**Comment by andris-sevcenko** [comment]
Why are you using a Redactor field for something that'll be used as a heading?

**Comment by dande-sun** [comment]
Did you get the answer?  I have the same problem, in my blog site, editors can fill the `Excerpt` field, but if they did not fill the excerpt, I need strip tags of the content field, and cut off it, because it is so long, let it fill the excerpt.

**Comment by brandonkelly** [comment, marked as answer]
Just [added](https://github.com/craftcms/cms/commit/17c933e41304942215fa34214c9a4d6afa0e0ee6) a `@stripTags` directive for the next release. Like [strip_tags()](https://www.php.net/manual/en/function.strip-tags.php), has an optional `allowed` argument, which can be set to an array of tag names that won’t get stripped.

I’ve also added a `@trim` directive for trimming leading/trailing whitespace.

```graphql
query {
  entries(section:"news" type:"article") {
    ...on news_article_Entry {
      myField @stripTags(allowed:["br"]) @trim
    }
  }
}
```

**Reply by brandonkelly** [reply]
Craft 4.5.4 is out with those new directives.
